packages/plugin-headings/web/src/toolbar/HeadingButton.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { mergeStyles } from 'wix-rich-content-common';
+import {
+  InlineToolbarButton,
+  EditorState,
+  HEADER_TYPE_MAP,
+  DEFAULT_HEADERS_DROPDOWN_OPTIONS,
+} from 'wix-rich-content-editor-common';
+import { RichUtils } from 'draft-js';
+import Modal from 'react-modal';
+import Panel from './HeadingPanel';
+import classNames from 'classnames';
+import styles from '../../statics/styles/headingButtonStyles.scss';
+
+export default class HeadingButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showPanel: false,
+      currentHeading: 'P',
+    };
+    this.styles = mergeStyles({ styles, theme: props.theme });
+    this.isFirstRender = true;
+  }
+
+  componentWillReceiveProps() {
+    const { isMobile } = this.props;
+    if (this.isFirstRender && isMobile) {
+      this.isFirstRender = false;
+    } else {
+      this.findCurrentHeading();
+    }
+  }
+
+  findCurrentHeading = () => {
+    const currentEditorState = this.props.getEditorState();
+    const selection = currentEditorState.getSelection();
+    const currentHeading = this.convertFontToHeadingName(
+      currentEditorState.getCurrentContent().blockMap.toJS()[selection.focusKey].type
+    );
+    this.setState({ currentHeading });
+  };
+
+  convertFontToHeadingName = font =>
+    Object.keys(HEADER_TYPE_MAP).find(key => HEADER_TYPE_MAP[key] === font);
+
+  openPanel = () => {
+    this.currentEditorState = this.oldEditorState = this.props.getEditorState();
+    this.selection = this.oldEditorState.getSelection();
+    const { bottom, left } = this.buttonRef.getBoundingClientRect();
+    this.props.setKeepOpen(true);
+    this.setState({ isPanelOpen: true, panelLeft: left - 15, panelTop: bottom });
+  };
+
+  closePanel = () => {
+    this.setState({ isPanelOpen: false });
+    this.props.setKeepOpen(false);
+  };
+
+  updateHeading = (font, heading) => {
```

Comment:
Check if this are correct 
```suggestion
  updateHeading = (type, element) => {
```

Reply:
done